pkg/pki/certmanagerpki/certmanager_user_test.go

```diff
@@ -103,6 +104,14 @@ func TestReconcileUserCertificate(t *testing.T) {
 		t.Error("Expected no error, got:", err)
 	}
 
+	// Test cert duration
+	user = newMockUser()
+	user.Spec.ExpirationSeconds = util.Int32Pointer(7200)
+
+	if _, err := manager.ReconcileUserCertificate(ctx, user, scheme.Scheme, clusterDomain); err != nil {
+		t.Error("Expected no error, got:", err)
+	}
```

Comment:
Instead of only verifying err, can we also verify the actual cert duration (IMO this is the most important part to verify here)?

Something like this as a reference:
```suggestion
	userCert, err := manager.ReconcileUserCertificate(ctx, user, scheme.Scheme, clusterDomain)
	if err != nil {
		t.Error("Expected no error, got:", err)
	}
	
	decodedCert, err := certutil.DecodeCertificate(userCert.Certificate)
	if err != nil {
		t.Error("Expected no error, got:", err)
	}
	
	if user.Spec.ExpirationSeconds != cert.NotAfter.Sub(cert.NotBefore).Seconds() {
	     t.Error("Expected cert duration is not the same as the actual cert duration, expected: %d, got: %d", user.Spec.ExpirationSeconds, cert.NotAfter.Sub(cert.NotBefore).Seconds())
	}
```

Reply:
Hi @panyuenlau, it seems if we compare the values, it is not working because time related fields are not populated properly in test env, NotAfter and NotBefore both have zero value of time.Time (i.e. 0001-01-01 00:00:00 +0000 UTC) and the difference is coming as zero. I think we have to check for no err for now. What do you think?

Reply:
Reverted the change.